Add note about sefcontext doing no restorecon

zeusttu opened the pull request:
To someone like me who is relatively new to SELinux, setting the `reload` option to `yes` might suggest that a `restorecon` is automatically executed after the `semanage` call, making the new file context effective immediately. I have found out that this is not the case and would like to clarify this to others.

+label: docsite_pr

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
sefcontext module

##### ANSIBLE VERSION
```
ansible 2.5.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/zeust/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, Aug  4 2017, 00:39:18) [GCC 4.8.5 20150623 (Red Hat 4.8.5-16)]
```

zeusttu committed:
Add note about sefcontext doing no restorecon

To someone like me who is relatively new to SELinux, setting the
"reload" option to yes might suggest that a restorecon is automatically
executed after the semanage call, making the new file context effective
immediately. I have found out that this is not the case and would like
to clarify this to others.

+label: docsite_pr

gundalow commented:
+label: docsite_pr

dagwieers commented:
I was pretty sure we already had a not like that ;-)

So it's true that the module does not do a restorecon, but that's intentional. In fact, it's advisable that you first add the new SELinux context before adding files (green field). So I would prefer this is reflected in the note.

Something like:

> The **sefcontext** module does not modify existing files to the new SELinux context(s), so it is advisable to first create the SELinux file contexts *before* creating files, or run **restorecon** manually for the existing files that require the new SELinux file contexts.

dagwieers commented:
@zeusttu No, but you can use these things like M(module_name), or I(parameter_name) or C(parameter_value). And also U(http://foo.bar/) or L(Title, http://foo.bar/).

See https://docs.ansible.com/ansible/2.5/dev_guide/developing_modules_documenting.html#formatting-functions

zeusttu committed:
Replace note by one suggested by reviewer

Reviewer dagwieers suggested a better notice text during review of my
original one, giving recommendations about what to do to actually get
the newly chosen SELinux context applied to the file.

zeusttu commented:
Thanks! I used the `C(executable)` to denote the name of an executable, because I guess an executable falls under the umbrella term "files", though I'm not 100% sure. Also I found no appropriate command in the referenced link to emphasise the word "before" so I left that one unemphasised. I hope it's alright like this, if not then tell me (or if you can you may also fix it yourself; I ticked a box when making this pull request that may or may not have given you this privilege) :slightly_smiling_face:

zeusttu commented:
Hmm looks like I will have to clone the repo locally so I can amend and force-push then. Will do.

zeusttu commented:
The problem was I credited you for your suggestion in my commit message but Ansibot complained that an @ mention in a commit message will somehow propagate to lots of places and should be edited out, so I had to amend my commit. I don't think Github's online editor allows that.

dagwieers commented:
@zeusttu Oh right, that was completely lost on me. You are right, you will have to pull that branch and interactively rebase it. Then you can update those commits and force-push them.

dagwieers reviewed (approved):
LGTM

dagwieers commented:
Perfect, thanks for your valuable contribution !

zeusttu commented:
You're welcome! 😁
